modules/hal_wch/Kconfig

```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2024 Dhiru Kholia
+#
+# SPDX-License-Identifier: Apache-2.0
```

Comment:
don't see why this file is here

Reply:
I am not sure but it seems to be required by the build system.

If I delete this file, the build process fails.

```
$ rm modules/hal_wch/Kconfig


(.venv) [user@zippy zephyr]$ west build -p always -b wch_ch32v003evt samples/basic/blinky
-- west build: making build dir /home/user/zephyrproject/zephyr/build pristine
-- west build: generating a build system
Loading Zephyr default modules (Zephyr base).
-- Application: /home/user/zephyrproject/zephyr/samples/basic/blinky
-- CMake version: 3.25.1
-- Found Python3: /home/user/zephyrproject/.venv/bin/python3 (found suitable version "3.11.7", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /home/user/.cache/zephyr
-- Zephyr version: 3.7.99 (/home/user/zephyrproject/zephyr)
-- Found west (found suitable version "1.2.0", minimum required is "0.14.0")
-- Board: wch_ch32v003evt, qualifiers: ch32v003
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.16.5 (/home/user/zephyr-sdk-0.16.5-1)
-- Found toolchain: zephyr 0.16.5 (/home/user/zephyr-sdk-0.16.5-1)
-- Found Dtc: /home/user/zephyr-sdk-0.16.5-1/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /home/user/zephyrproject/zephyr/boards/wch/ch32v003evt/wch_ch32v003evt.dts
-- Generated zephyr.dts: /home/user/zephyrproject/zephyr/build/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /home/user/zephyrproject/zephyr/build/zephyr/include/generated/zephyr/devicetree_generated.h
-- Including generated dts.cmake file: /home/user/zephyrproject/zephyr/build/zephyr/dts.cmake
Parsing /home/user/zephyrproject/zephyr/Kconfig
/home/user/zephyrproject/zephyr/scripts/kconfig/kconfig.py: build/Kconfig/Kconfig.modules:134: Could not open '/home/user/zephyrproject/zephyr/' (in 'osource "$(ZEPHYR_HAL_WCH_KCONFIG)"') (EISDIR: Is a directory)
CMake Error at /home/user/zephyrproject/zephyr/cmake/modules/kconfig.cmake:389 (message):
  command failed with return code: 1
Call Stack (most recent call first):
  /home/user/zephyrproject/zephyr/cmake/modules/zephyr_default.cmake:133 (include)
  /home/user/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:66 (include)
  /home/user/zephyrproject/zephyr/share/zephyr-package/cmake/ZephyrConfig.cmake:92 (include_boilerplate)
  CMakeLists.txt:4 (find_package)
```

Reply:
```
/home/user/zephyrproject/zephyr/scripts/kconfig/kconfig.py: build/Kconfig/Kconfig.modules:134: Could not open '/home/user/zephyrproject/zephyr/' (in 'osource "$(ZEPHYR_HAL_WCH_KCONFIG)"') (EISDIR: Is a directory)
```

the module system really wants a file to be there